libvast/builtins/formats/zeek_tsv.cpp

```diff
@@ -0,0 +1,757 @@
+    };
+  }
+
+  auto operator()(const int64_type&) const -> result_type {
+    return parsers::i64->*[](int64_t x) {
+      return int64_t{x};
+    };
+  }
+
+  auto operator()(const uint64_type&) const -> result_type {
+    return parsers::u64->*[](uint64_t x) {
+      return x;
+    };
+  }
+
+  auto operator()(const time_type&) const -> result_type {
+    return parsers::real->*[](double x) {
+      auto i = std::chrono::duration_cast<duration>(double_seconds(x));
+      return time{i};
+    };
+  }
+
+  auto operator()(const duration_type&) const -> result_type {
+    return parsers::real->*[](double x) {
+      return std::chrono::duration_cast<duration>(double_seconds(x));
+    };
+  }
+
+  auto operator()(const string_type&) const -> result_type {
+    if (set_sep_.empty())
+      return +parsers::any->*[](std::string x) {
+        return detail::byte_unescape(x);
+      };
+    return +(parsers::any - set_sep_)->*[](std::string x) {
+      return detail::byte_unescape(x);
+    };
+  }
+
+  auto operator()(const ip_type&) const -> result_type {
+    return parsers::ip->*[](ip x) {
+      return x;
+    };
+  }
+
+  auto operator()(const subnet_type&) const -> result_type {
+    return parsers::net->*[](subnet x) {
+      return x;
+    };
+  }
+
+  auto operator()(const list_type& t) const -> result_type {
+    return (caf::visit(*this, t.value_type()) % set_sep_)
+             ->*[](std::vector<Attribute> x) {
+                   return list(std::move(x));
+                 };
+  }
+
+  const std::string& set_sep_;
+};
+
+/// Constructs a Zeek data parser from a type and set separator.
+template <class Iterator, class Attribute = data>
+auto make_zeek_parser(const type& t, const std::string& set_sep = ",")
+  -> rule<Iterator, Attribute> {
+  rule<Iterator, Attribute> r;
+  auto sep = is_container(t) ? set_sep : "";
+  return caf::visit(zeek_parser_factory<Iterator, Attribute>{sep}, t);
+}
+
+// Creates a VAST type from an ASCII Zeek type in a log header.
+auto parse_type(std::string_view zeek_type) -> caf::expected<type> {
+  type t;
+  if (zeek_type == "enum" or zeek_type == "string" or zeek_type == "file"
+      or zeek_type == "pattern")
+    t = type{string_type{}};
+  else if (zeek_type == "bool")
+    t = type{bool_type{}};
+  else if (zeek_type == "int")
+    t = type{int64_type{}};
+  else if (zeek_type == "count")
+    t = type{uint64_type{}};
+  else if (zeek_type == "double")
+    t = type{double_type{}};
+  else if (zeek_type == "time")
+    t = type{time_type{}};
+  else if (zeek_type == "interval")
+    t = type{duration_type{}};
+  else if (zeek_type == "addr")
+    t = type{ip_type{}};
+  else if (zeek_type == "subnet")
+    t = type{subnet_type{}};
+  else if (zeek_type == "port")
+    // FIXME: once we ship with builtin type aliases, we should reference the
+    // port alias type here. Until then, we create the alias manually.
+    // See also:
+    // - src/format/pcap.cpp
+    t = type{"port", uint64_type{}};
```

Comment:
I wonder what's our current thinking here. (cc @dominiklohmann)

We could also provide "global" lookup functions to query the type registry, in order to look for `port` type and only fall back to hand-creating a `port` type if not found.

Reply:
We don't want to change this now, that's too far away from our current trajectory. What we have works good enough for now. If we ever want to change this, we can replace this in the future.